edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -13,47 +13,58 @@ namespace Microsoft.Azure.Devices.Edge.Hub.Core
     using Microsoft.Extensions.Logging;
     using Newtonsoft.Json;
     using Nito.AsyncEx;
-    using AsyncLock = Microsoft.Azure.Devices.Edge.Util.Concurrency.AsyncLock;
 
     public sealed class DeviceScopeIdentitiesCache : IDeviceScopeIdentitiesCache
     {
         readonly IServiceProxy serviceProxy;
         readonly IKeyValueStore<string, string> encryptedStore;
-        readonly AsyncLock cacheLock = new AsyncLock();
-        readonly IDictionary<string, StoredServiceIdentity> serviceIdentityCache;
+        readonly IServiceIdentityTree serviceIdentityTree;
         readonly Timer refreshCacheTimer;
         readonly TimeSpan refreshRate;
         readonly AsyncAutoResetEvent refreshCacheSignal = new AsyncAutoResetEvent();
         readonly object refreshCacheLock = new object();
+        readonly bool nestedEdgeEnabled;
```

Comment:
deviceidentititescache should not have nestedEdgeEnabled, serviceProxy internally should return the iterator

Reply:
talked offline, moved this to ServiceProxy